integration_tests/system_test.go

```diff
@@ -0,0 +1,125 @@
+/*
+ Copyright 2022 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package integration_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/openconfig/ondatra"
+)
+
+// TestHostname verifies that the hostname configuration paths can be read,
+// updated, and deleted.
+//
+// config_path:/system/config/hostname
+// telemetry_path:/system/state/hostname
+func TestHostname(t *testing.T) {
+	testCases := []struct {
+		description string
+		hostname    string
+	}{
+		{"15 Letters", "abcdefghijkmnop"},
+		{"15 Numbers", "123456789012345"},
+		{"Single Character", "x"},
+		{"Periods", "test.name.example"},
+		{"63 Characters", "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"},
+	}
+
+	dut := ondatra.DUT(t, "dut")
+
+	for _, testCase := range testCases {
+		t.Run(testCase.description, func(t *testing.T) {
+			config := dut.Config().System().Hostname()
```

Comment:
here and below: use the new telemetry library. We want to delete the old one soon https://github.com/openconfig/ondatra/blob/main/gnmi/gnmi.go

Reply:
This library doesn't seem to have an API to get config paths.

Reply:
the library is ygnmi. gnmi.Get[string](t, ocpath.Root().System().Hostname().Config())

Reply:
Ah so it can't infer T correctly since the generic type is an interface. Perhaps ygnmi should have a GetConfig to help with this?

Reply:
we don't because if the type inference get fixed it will easier to remove all the newly unnecessary types than go and change all the instances of GetConfig back Get